File: python/examples/reviews/v1beta/get_merchant_review_sample.py
# -*- coding: utf-8 -*-
# Copyright 2025 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
# [START merchantapi_get_merchant_review]
"""This class demonstrates how to get a Merchant review."""

from examples.authentication import configuration
from examples.authentication import generate_user_credentials
from google.shopping.merchant_reviews_v1beta import GetMerchantReviewRequest
from google.shopping.merchant_reviews_v1beta import MerchantReviewsServiceClient


def get_merchant_review(account_id: str, merchant_review_id: str) -> None:
  """Gets a merchant review from the given account.

  Args:
    account_id: The ID of the Merchant Center account.
    merchant_review_id: The ID of the merchant review to retrieve.
  """
  # Gets OAuth credentials.
  credentials = generate_user_credentials.main()

  # Creates a client.
  client = MerchantReviewsServiceClient(credentials=credentials)

  # The name of the review to retrieve.
  # Format: accounts/{account}/merchantReviews/{merchant_review}
  name = f"accounts/{account_id}/merchantReviews/{merchant_review_id}"

  # Creates the request.
  request = GetMerchantReviewRequest(name=name)

  # Makes the request and catches and prints any error messages.
  try:
    print("Sending get merchant review request:")
    response = client.get_merchant_review(request=request)
    print("Merchant review retrieved successfully:")
    print(response)
  except RuntimeError as e:
    print(e)


if __name__ == "__main__":
  # Gets the merchant account ID from the user.
  merchant_account_id = configuration.Configuration().read_merchant_info()

  # The review ID is the last segment of the `name` field of the
  # `MerchantReview` resource. For example, if the `name` is
  # `accounts/12345/merchantReviews/67890`, the review ID is `67890`.
  review_id = "YOUR_MERCHANT_REVIEW_ID"

  get_merchant_review(merchant_account_id, review_id)


# [END merchantapi_get_merchant_review]

File: php/examples/reviews/v1/GetMerchantReviewSample.php
<?php
/**
 * Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

require_once __DIR__ . '/../../../vendor/autoload.php';
require_once __DIR__ . '/../../Authentication/Authentication.php';
require_once __DIR__ . '/../../Authentication/Config.php';
// [START merchantapi_get_merchant_review]
use Google\ApiCore\ApiException;
use Google\Shopping\Merchant\Reviews\V1beta\Client\MerchantReviewsServiceClient;
use Google\Shopping\Merchant\Reviews\V1beta\GetMerchantReviewRequest;

/**
 * This class demonstrates how to get a Merchant review.
 */
class GetMerchantReviewSample
{
    private const MERCHANT_REVIEW_ID = 'YOUR_MERCHANT_REVIEW_ID';

    /**
     * Retrieves a merchant review from your Merchant Center account.
     *
     * @param array $config The configuration data for authentication and account ID.
     * @param string $merchantReviewId The ID of the merchant review to retrieve.
     */
    public static function getMerchantReviewSample(array $config, string $merchantReviewId): void
    {
        // Gets the OAuth credentials to make the request.
        $credentials = Authentication::useServiceAccountOrTokenFile();

        // Creates options config containing credentials for the client to use.
        $options = ['credentials' => $credentials];

        // Creates a client.
        $merchantReviewsServiceClient = new MerchantReviewsServiceClient($options);

        // The name of the merchant review to retrieve.
        // Format: accounts/{account}/merchantReviews/{merchant_review}
        $name = sprintf(
            'accounts/%s/merchantReviews/%s',
            $config['accountId'],
            $merchantReviewId
        );

        // Creates the request message.
        $request = (new GetMerchantReviewRequest())
            ->setName($name);

        // Calls the API and catches and prints any network failures/errors.
        try {
            printf("Sending get merchant review request:%s", PHP_EOL);
            $response = $merchantReviewsServiceClient->getMerchantReview($request);
            printf("Merchant review retrieved successfully:%s", PHP_EOL);
            printf("%s%s", $response->getName(), PHP_EOL);
        } catch (ApiException $e) {
            print $e->getMessage() . PHP_EOL;
        }
    }

    /**
     * Helper to execute the sample.
     */
    public function callSample(): void
    {
        $config = Config::generateConfig();
        self::getMerchantReviewSample($config, self::MERCHANT_REVIEW_ID);
    }
}

// Run the script.
$sample = new GetMerchantReviewSample();
$sample->callSample();
// [END merchantapi_get_merchant_review]

